File: 02-Python/05-leap-year.py
#!/usr/bin/python

while True:

    your_year = int(input("Enter a year and check whether it is a leap year or not!\n"))

    def is_leap_year(year):
        if year % 4 == 0:
            if year % 100 == 0:
                if year % 400 == 0:
                    print("\n" + str(year) + " is a leap year\n")
                else:
                    print("\n" + str(year) + " is not a leap year\n")
            else:
                print("\n" + str(year) + " is a leap year\n")
        else:
            print("\n" + str(year) + " is not a leap year\n")

   
    is_leap_year(your_year)


    if input("Do you want to check another year? Y/N: ").lower() not in {"y", "yes"}:
        break

File: 01-JavaScript/05-leap-year.js
const prompt = require("prompt-sync")({ sigint: true });

const yourYear = prompt("Write a year and check if it is leap or not (e.g. 2010): ");

function isLeapYear(year) {
    const yearNum = parseInt(year);

    if (yearNum % 4 === 0) {
        if (yearNum % 100 === 0) {
            if (yearNum % 400 === 0) {
                return "Leap year.";
            } else {
                return "Not leap year.";
            }
        } else {
            return "Leap year.";
        }
    } else {
        return "Not leap year.";
    }
}

const leapOrNot = isLeapYear(yourYear);
console.log(leapOrNot);